packages/wallet-adapter-core/src/AIP62StandardWallets/WalletStandard.ts

```diff
@@ -0,0 +1,176 @@
+import {
+  UserResponse,
+  AptosSignTransactionOutput,
+  AptosSignMessageOutput,
+  AptosSignMessageInput,
+  AptosWallet,
+  UserResponseStatus,
+  AptosSignAndSubmitTransactionOutput,
+  AccountInfo as StandardAccountInfo,
+} from "@aptos-labs/wallet-standard";
+import {
+  AnyRawTransaction,
+  PendingTransactionResponse,
+  Aptos,
+  MultiEd25519Signature,
+  MultiEd25519PublicKey,
+} from "@aptos-labs/ts-sdk";
+
+import { WalletReadyState } from "../constants";
+import {
+  WalletConnectionError,
+  WalletSignAndSubmitMessageError,
+  WalletSignMessageAndVerifyError,
+  WalletSignMessageError,
+} from "../error";
+import {
+  AccountInfo,
+  InputTransactionData,
+  Wallet,
+} from "../LegacyWalletPlugins/types";
+import { generalizedErrorMessage } from "../utils";
+
+export type AptosStandardWallet = AptosWallet & {
+  readyState?: WalletReadyState;
+};
+
+export class WalletStandardCore {
+  /**
+   * Signs and submits a transaction to chain
+   *
+   * @param transactionInput InputTransactionData
+   * @returns PendingTransactionResponse
+   */
+  async signAndSubmitTransaction(
+    transactionInput: InputTransactionData,
+    aptos: Aptos,
+    account: AccountInfo,
+    wallet: Wallet
```

Comment:
we're still passing a "legacy" wallet to the new WalletStandardCore?

Reply:
it is not a "legacy" wallet but a wallet adapter Wallet type, this is to not introduce any breaking changes for dapps

Reply:
we could do it differently and
- convert `wallet` to `AptosStandardWallet`
- Pass it to the new standard `signAndSubmitTransaction` function
- and then do `wallet.features["aptos:signAndSubmitTransaction"]?.signAndSubmitTransaction(
        transaction
      )`

but we already converted each `AptosStandardWallet` wallet into legacy `Wallet` type so we kind of convert it now back.

I think once we deprecate any legacy wallet plugin, we will have to change adapter types and therefore input and output types - that would introduce a breaking change